cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -218,37 +234,57 @@ func (s *Supervisor) loadConfig(configFile string) error {
 	return nil
 }
 
+// getBootstrapInfo obtains the Collector's agent description by
+// starting a Collector with a specific config that only start's
+// an OpAMP extension, obtains the agent description, then
+// shuts down the Collector. This only needs to happen
+// once per Collector binary.
 func (s *Supervisor) getBootstrapInfo() (err error) {
 	port, err := s.findRandomPort()
 	if err != nil {
 		return err
 	}
 
-	supervisorPort, err := s.findRandomPort()
+	s.opampServerPort, err = s.findRandomPort()
 	if err != nil {
 		return err
 	}
 
-	var cfg bytes.Buffer
+	var k = koanf.New(".")
 
+	var cfg bytes.Buffer
 	err = s.bootstrapTemplate.Execute(&cfg, map[string]any{
 		"EndpointPort":   port,
```

Comment:
Can we rename `EndpointPort` to something more descriptive? It is now confusing that we have 2 ports that we obtain randomly and pass to the bootstrap template.

Reply:
We were able to get rid of this now that we use the nop components during bootstrapping. The only port left is the one used for the OpAMP connection.